Octokit/Helpers/ReferenceExtensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Octokit.Helpers
+{
+    /// <summary>
+    /// Represents operations to simplify working with references
+    /// </summary>
+    public static class ReferenceExtensions
+    {
+        /// <summary>
+        /// Creates a branch, based off the branch specified.
+        /// </summary>
+        /// <param name="referencesClient">The <see cref="IReferencesClient" /> this method extends</param>
+        /// <param name="owner">The owner of the repository.</param>
+        /// <param name="name">The name of the repository.</param>
+        /// <param name="branchName">The new branch name</param>
+        /// <param name="baseReference">The <see cref="Reference" /> to base the branch from</param>
+        public static async Task<Reference> CreateBranch(this IReferencesClient referencesClient, string owner, string name, string branchName, Reference baseReference)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(name, "name");
+            Ensure.ArgumentNotNullOrEmptyString(branchName, "branchName");
+            Ensure.ArgumentNotNull(baseReference, "baseReference");
+
+            if (branchName.StartsWith("refs/heads"))
+            {
+                throw new ArgumentException("The branchName seems to be misformed. Try using the Create method");
```

Comment:
Close! :smile:

I was thinking something more specific.

```csharp
throw new ArgumentException(String.Format(CultureInfo.InvariantCulture, "The specified branchName '{0}' appears to be a ref name and not a branch name because it starts with the string 'refs/heads'. Either specify just the branch name or use the Create method if you need to specify the full ref name", branchName));
```